whisper/shhapi/api.go

```diff
@@ -367,32 +367,32 @@ func (api *PublicWhisperAPI) Post(args PostArgs) error {
 }
 
 type PostArgs struct {
-	TTL      uint32              `json:"ttl"`
-	From     rpc.HexBytes        `json:"from"`
-	To       rpc.HexBytes        `json:"to"`
-	KeyName  string              `json:"keyname"`
-	Topic    whisperv5.TopicType `json:"topic"`
-	Padding  rpc.HexBytes        `json:"padding"`
-	Payload  rpc.HexBytes        `json:"payload"`
-	WorkTime uint32              `json:"worktime"`
-	PoW      float64             `json:"pow"`
-	FilterID int                 `json:"filter"`
-	PeerID   rpc.HexBytes        `json:"directP2P"`
+	TTL      uint32
+	From     string
+	To       string
+	KeyName  string
+	Topic    whisperv5.TopicType
+	Padding  rpc.HexBytes
+	Payload  rpc.HexBytes
+	WorkTime uint32
+	PoW      float64
+	FilterID int
+	PeerID   rpc.HexBytes
 }
 
 func (args *PostArgs) UnmarshalJSON(data []byte) (err error) {
```

Comment:
This seem a little complicated way to achieve a custom serialised filterID. Also -1 values are very suspicious.